wire-java-generator/src/main/java/com/squareup/wire/java/ProfileLoader.java

```diff
@@ -21,51 +21,69 @@
 import com.google.common.io.Closer;
 import com.squareup.wire.java.internal.ProfileFileElement;
 import com.squareup.wire.java.internal.ProfileParser;
+import com.squareup.wire.java.internal.TypeConfigElement;
 import com.squareup.wire.schema.Location;
 import com.squareup.wire.schema.ProtoFile;
+import com.squareup.wire.schema.ProtoType;
 import com.squareup.wire.schema.Schema;
+import com.squareup.wire.schema.SchemaException;
+import com.squareup.wire.schema.Type;
 import java.io.IOException;
 import java.nio.file.FileSystem;
 import java.nio.file.FileSystems;
 import java.nio.file.Files;
 import java.nio.file.Path;
-import java.nio.file.Paths;
+import java.util.ArrayList;
 import java.util.Collection;
 import java.util.LinkedHashSet;
+import java.util.List;
 import java.util.Map;
 import java.util.Set;
 import okio.Okio;
 import okio.Source;
 
+import static com.google.common.base.Preconditions.checkState;
 import static com.google.common.collect.Iterables.getOnlyElement;
 
 /**
  * Load files with the {@code .wire} suffix for a schema. The file name is the profile name; for
  * example the {@code android} profile loads {@code android.wire} files.
  */
 public final class ProfileLoader {
+  private final FileSystem fileSystem;
   private final String name;
-  private final Set<Location> protoLocations = new LinkedHashSet<>();
+  private Schema schema;
 
-  public ProfileLoader(String name) {
+  public ProfileLoader(FileSystem fileSystem, String name) {
+    this.fileSystem = fileSystem;
     this.name = name;
   }
 
-  public ProfileLoader addSchema(Schema schema) {
-    for (ProtoFile protoFile : schema.protoFiles()) {
-      protoLocations.add(protoFile.location());
-    }
-    return this;
+  public ProfileLoader(String name) {
+    this(FileSystems.getDefault(), name);
   }
 
-  public ProfileLoader addProtoLocation(Location location) {
-    protoLocations.add(location);
+  public ProfileLoader schema(Schema schema) {
+    checkState(this.schema == null);
```

Comment:
should be `checkNotNull` or perhaps `checkArgument`. Getting an ISE for passing `null` is weird.